gitpod-io/gitpod
# [ide] use code flag to install extensions

## Description
Track readiness for `content` `ide` `ide-desktop` ready

Upstream vscode has provided extension install with flag `--install-extension` and there are some bugs with our installation code 
~~🐛 Remove extensionIdRegex for extensions that install via file path~~
🐛 Support extensions that their id starts with http or https

It's a Golang copy of [these codes](https://github.com/gitpod-io/openvscode-server/blob/96d57b52465f59b5c0ff650d2413e99451106f83/src/vs/gitpod/node/customServerIntegration.ts#L123-L161)

And part of extension host timeout https://github.com/gitpod-io/gitpod/issues/7863 fixes, more detail see [internal slack](https://gitpod.slack.com/archives/C01KGM9BH54/p1645044794473059)

## Related Issue(s)

Relates #7863, fixes https://github.com/gitpod-io/gitpod/issues/7409

## How to test

For test, we use this version of vscode which remove our custom extensions install only https://github.com/gitpod-io/openvscode-server/commit/6d0747dbcd685cfe47251d56a452b619c9da444b

Just remove support of install with file path

1. Make sure to use insider vscode https://hw-code-flag.staging.gitpod-dev.com/preferences
2. Open a workspace, you can use this repo https://hw-code-flag.staging.gitpod-dev.com/#github.com/mustard-mh/test/tree/vs/exthost-no-file
   <img width="991" alt="image" src="https://user-images.githubusercontent.com/20944364/154689220-e4f305e3-4808-46ad-97dc-f2696f203875.png">
3. Click `Extensions` button of vscode check if all your extensions installed
4. Run `ps -efj fww` should have a process like 
  ```
sh /ide/bin/gitpod-code --start-server --install-builtin-extension github.vscode-pull-request-github --install-extension bierner.color-info --install-extension vscodevim.vim@1.21.10 --install-extension magicstack.magicpython --install-extension formulahendry.code-runner --port 23000 --host 0.0.0.0 --without-connection-token --server-data-dir /workspace/.vscode-remote
  ```
5. Go to [`segment.com`](https://app.segment.com/gitpod/sources/staging_untrusted/debugger) search with `supervisor_readiness` to see event tracking


## Release Notes

```release-note
NONE
```

## Documentation

/werft analytics=segment|TEZnsG4QbLSxLfHfNieLYGF4cDwyFWoe
